controllers/release/adapter_test.go

```diff
@@ -1304,25 +1358,81 @@ var _ = Describe("Release adapter", Ordered, func() {
 		})
 
-		It("returns an error if automated label is present but is not set in release status", func() {
-			adapter.release.Labels = map[string]string{
-				metadata.AutomatedLabel: "true",
-			}
+		It("should return valid and no error if all the resources are found", func() {
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ProcessingResourcesContextKey,
+					Resource: &loader.ProcessingResources{
+						EnterpriseContractConfigMap: enterpriseContractConfigMap,
+						EnterpriseContractPolicy:    enterpriseContractPolicy,
+						ReleasePlanAdmission:        releasePlanAdmission,
+						ReleaseStrategy:             releaseStrategy,
+						Snapshot:                    snapshot,
+					},
+				},
+			})
 
-			err := adapter.validateAuthor(releasePlan)
-			Expect(err).NotTo(BeNil())
-			Expect(err.Error()).To(Equal("automated not set in status for automated release"))
+			valid, err := adapter.validateProcessingResources()
+			Expect(valid).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
 		})
 
-		It("validates the author", func() {
-			adapter.release.Labels = map[string]string{
-				metadata.AuthorLabel: "user",
-			}
-			err := adapter.validateAuthor(releasePlan)
-			Expect(err).To(BeNil())
-			Expect(adapter.release.Status.Attribution.Author).To(Equal("user"))
+		It("should return invalid and no error if any of the resources are not found", func() {
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ProcessingResourcesContextKey,
+					Err:        errors.NewNotFound(schema.GroupResource{}, ""),
+				},
+			})
+
+			valid, err := adapter.validateProcessingResources()
+			Expect(!valid).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should return invalid and no error if the ReleasePlanAdmission is found to be disabled", func() {
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ReleasePlanAdmissionContextKey,
+					Err:        fmt.Errorf("auto-release label set to false"),
+				},
+			})
+
+			valid, err := adapter.validateProcessingResources()
+			Expect(!valid).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should return invalid and no error if multiple ReleasePlanAdmissions exist", func() {
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ReleasePlanAdmissionContextKey,
+					Err:        fmt.Errorf("multiple ReleasePlanAdmissions found"),
+				},
+			})
+
+			valid, err := adapter.validateProcessingResources()
+			Expect(!valid).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should return invalid and an error if some other type of error occurs", func() {
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ProcessingResourcesContextKey,
+					Err:        fmt.Errorf("internal error"),
+					Resource: &loader.ProcessingResources{
```

Comment:
It seems that is not properly aligned, but maybe it's just in GitHub. I noticed something similar today.

Reply:
that is how the ide wants it